scala/scala.bzl

```diff
@@ -162,9 +162,31 @@ def _compile(ctx, cjars, dep_srcjars, buildijar):
     all_srcjars = set(srcjars + list(dep_srcjars))
     # look for any plugins:
     plugins = _collect_plugin_paths(ctx.attr.plugins)
+
+    if (hasattr(ctx.attr, 'enable_dependency_analyzer')
+        and ctx.attr.enable_dependency_analyzer
+        and hasattr(ctx.attr, 'dependency_analyzer_plugin')):
+      enable_dependency_analyzer = ctx.attr.enable_dependency_analyzer
+      dep_plugin = ctx.attr.dependency_analyzer_plugin
+      plugins += [f.path for f in dep_plugin.files]
+      dependency_analyzer_plugin_jars = ctx.files.dependency_analyzer_plugin
+    else:
+      enable_dependency_analyzer = False
+      dependency_analyzer_plugin_jars = []
+
     plugin_arg = ",".join(list(plugins))
 
-    compiler_classpath = ":".join([j.path for j in cjars])
+    all_jars = cjars + rjars
+    compiler_classpath = ":".join([j.path for j in all_jars])
+    direct_jars = ",".join([j.path for j in cjars])
+
+    valid_jar_paths = []
```

Comment:
This was added in order to pass non-related failing tests.

On retrospect, this was incorrect and there was a missing test on 'file' deps, e.g. `@com_google_guava_guava_21_0//jar:file`.
added a test, and deleted these lines